### PR TITLE
feat: remove references to introspection-engine and PRISMA_INTROSPECTION_ENGINE

### PR DESCRIPTION
**Pull request by jkomyno**
DO NOT MERGE UNTIL `prisma@4.10.0` is ready to be released.

## Describe this PR

Follow-up for https://github.com/prisma/prisma/pull/17301.
We've stopped using the `introspection-engine` binary in `prisma@4.9.0-dev.*`. As a result, the `PRISMA_INTROSPECTION_ENGINE` environment variable is now ignored. I've also removed mentions to `introspection-engine` in the snippets reporting the output of `prisma version`.

## Changes

Remove references to `PRISMA_INTROSPECTION_ENGINE` and `introspection-engine`.

## What issue does this fix?

Contributes to https://github.com/prisma/prisma/issues/17300.

### CODE REVIEW

**Review by ruheni — APPROVED** [review]

**Comment by janpio** [comment]
Same comment as for https://github.com/prisma/docs/pull/4292#issuecomment-1420486905:

Where did the documentation of the env var move for users of older Prisma versions?
We only have one docs website, so we need to make sure it also works for other versions.
In this case we could for example add a "In older versions" headline that has a minimal mention of what the env var used to to and be with info when we removed it.

**Comment by janpio** [comment]
And also same problem for `content/300-guides/200-deployment/110-deployment-guides/800-use-prisma-with-serverless-framework.mdx`

File: content/200-concepts/100-components/08-prisma-engines/index.mdx [hunk 1]
```diff
@@ -40,6 +40,12 @@ Use the following environment variables to specify custom locations for your bin
 - [`PRISMA_INTROSPECTION_ENGINE_BINARY`](/reference/api-reference/environment-variables-reference#prisma_introspection_engine_binary) <span class="api"></span> (Introspection engine)
 - [`PRISMA_FMT_BINARY`](/reference/api-reference/environment-variables-reference#prisma_fmt_binary) <span class="api"></span> (for `npx prisma format`)
 
+<Admonition type="warning">
+
+- The Introspection Engine is served by the Migration Engine from [4.9.0](https://github.com/prisma/prisma/releases/tag/4.9.0). Therefore, the `PRISMA_INTROSPECTION_ENGINE` environment variable will not be used.
+
+</Admonition>
```
Comment (janpio) [inline content/200-concepts/100-components/08-prisma-engines/index.mdx]: Here we can just remove it - only in the reference old users need to find a mention of it.

Comment (ruheni) [inline content/200-concepts/100-components/08-prisma-engines/index.mdx]:
```suggestion

```

**Pull request merged**